packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -352,6 +352,13 @@ contract Bridge is EssentialContract, IBridge {
             revert B_NON_RETRIABLE();
         }
 
+        // We check _message.gasLimit > gasleft() to make sure we not only need to bridge invocation
+        // call to succeeed, we also need it to succeed with a gas limit no smaller than the
+        // message's gasLimit.
+        if (_message.gasLimit != 0 && _message.gasLimit > gasleft()) {
```

Comment:
Dani, I think the 1/64 also apply here. Assuming `gasleft() == gasLimit`, then only `63/64 * gasleft()` can be forward to the destination contract.

So maybe we check `_message.gasLimit > (63* gasleft())>> 6)`

Reply:
If this make sense, maybe we extract a small function for the 1/64 rule.

Reply:
Okay, i can do, but there is a reason i did not put it there. Because simple if that is the case and code reverts because there is no more gas to execute thre resot of the functions (emit event, etc.) then there is no effect, right ? Simply people just retry with higher gas. Not really safety critical and we dont expect that to happen (too often) ?

Reply:
if gasleft() == gasLimit, then 63*gasLimit/64 gas will be forwarded to the dest app to call the function, but we do not want this call to succeed.

Reply:
added the check. (Also not created a separate function but inlined, since it is a lil bit cheaper)